lib/doorkeeper/oauth/refresh_token_request.rb

```diff
@@ -67,7 +67,11 @@ def validate_client_match
 
       def validate_scope
         if @original_scopes.present?
-          ScopeChecker.valid?(@original_scopes, refresh_token.scopes)
+          ScopeChecker.valid?(
+              @original_scopes,
+              refresh_token.scopes,
+              @client.try(:scopes)
```

Comment:
You can use the `client` getter (no need of the at sign).
Also, indentation is weird (two additional spaces).